### PR TITLE
chore(scala-style) : add Scala style and formatting resources file

### PR DESCRIPTION
**Pull request by amarouni**
Add Scala style file used by DataStreams and DataCatalog.

### CODE REVIEW

File: tools-scala-style/README.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
+Scala-Style : http://www.scalastyle.org/
+
+The Scala style xml is used with the Scala style SBT plugin : http://www.scalastyle.org/sbt.html
```
Comment (sgandon) [inline tools-scala-style/README.md]: Could you elaborate on how to set this up for a project starting from scratch please.

**Comment by amarouni** [comment]
@tricqueb The `data-streams-settings` plugin is really a specific data-streams/dataset plugin, we declare lots of project specific dependencies. But I get what you mean, maybe we just should create a basic scala Talend plugin without any specific project stuff, then build project specific stuff on top of it.

What do you mean by compiler options ? I have some SBT snippets for both plugins.

**Comment by tricqueb** [comment]
:+1: yes indeed I think we truly need to have a shared Talend plugin that will ship all of this. You are right about data-stream-settings, it might too specific. Let's keep this that way, and we will update possibilities when we'll have a sharable Talend plugin.
By compiler options, I meant generic compiler options to put in global.scala or in talend plugin like `        "-Ywarn-unused"`. But it might not be the right place ?

**Comment by amarouni** [comment]
@tricqueb You can update the Readme with a specific section on recommended compiler options, that would be a good addition.

**Pull request merged**